packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -121,16 +135,103 @@ const SideNavFooterContainer = styled("div", {
   },
 }));
 
+const getHasScrollableContent = (scrollableContainer: HTMLElement) =>
+  scrollableContainer.scrollHeight > scrollableContainer.clientHeight;
+
 const SideNav = ({
   navHeaderText,
   isCollapsible,
   onCollapse,
   onExpand,
   sideNavItems,
+  expandedWidth = "300px",
   footerItems,
+  footerComponent,
+  logo,
 }: SideNavProps) => {
   const [isSideNavCollapsed, setSideNavCollapsed] = useState(false);
   const odysseyDesignTokens = useOdysseyDesignTokens();
+  const [isContentScrollable, setIsContentScrollable] = useState(false);
+  const [hasContentScrolled, setHasContentScrolled] = useState(false);
+  const scrollableContentRef = useRef<HTMLUListElement>(null);
+  const resizeObserverRef = useRef<ResizeObserver | null>(null);
+  const intersectionObserverRef = useRef<IntersectionObserver | null>(null);
+
+  useEffect(() => {
+    const setWhetherContentIsScrollable = () => {
+      if (
+        scrollableContentRef.current &&
+        scrollableContentRef.current.parentElement
+      ) {
+        setIsContentScrollable(
+          getHasScrollableContent(scrollableContentRef.current.parentElement),
+        );
+      }
+    };
+
+    // If the window is resized, we may need to re-determine if the scrollable container has overflow
+    // Setup a ResizeObserver to know if the size of the scrollableContent changes
+    let resizeObserverDebounceTimer: ReturnType<typeof requestAnimationFrame>;
+    if (!resizeObserverRef.current) {
+      resizeObserverRef.current = new ResizeObserver(() => {
+        cancelAnimationFrame(resizeObserverDebounceTimer);
+        resizeObserverDebounceTimer = requestAnimationFrame(
+          setWhetherContentIsScrollable,
+        );
+      });
+    }
+
+    if (resizeObserverRef.current && scrollableContentRef.current) {
+      // Observe the <ul> itself (in case it changes size due to the content expanding)
+      resizeObserverRef.current.observe(scrollableContentRef.current);
+      if (scrollableContentRef.current.parentElement) {
+        // ALSO observe the parent (<SideNavScrollableContainer>) in case the window resizes
+        resizeObserverRef.current.observe(
+          scrollableContentRef.current.parentElement,
+        );
+      }
+    }
+
+    // Determine if the scrollable container has overflow or not on load
+    setWhetherContentIsScrollable();
```

Comment:
Bit of a nit but, can we rename this to something like `checkIfContentIsScrollable` or something like that?

Reply:
Yeah, I seen that as well. This is a weird one to name. We can just leave it as is. I don't feel super strongly about it.

Reply:
I'm fine with the `set` prefix, I just don't like explaining the `if` check in the function name.

I'm assuming `setIsScrollable` is taken already?

We're not always setting the value. That's the hard part. This function is conditionally setting the value.

I wonder if we should be doing a mutation with this function in the first place.

Maybe instead it should be like this:
`setIsContentScrollable(computeIsContentScrollable()))`

The text `compute` says we're doing some work to get if the content is scrollable. If we used the prefix `get`, it would seem like this simply gets the value out of the object.

I like having the `setIsContentScrollable` since mutations are generally frowned upon, and we don't run into a "two things are named nearly the same and perform almost the same function" situation.

What do you think?

Reply:
after discussing with Kevin synchronously, have simply changed the name to `updateIsContentScrollable`